src/main/scala/nodes/learning/PCA.scala

```diff
@@ -47,11 +48,111 @@ case class BatchPCATransformer(pcaMat: DenseMatrix[Float]) extends Transformer[D
  *
  * @param dims Dimensions to reduce input dataset to.
  */
-case class ColumnPCAEstimator(dims: Int) extends Estimator[DenseMatrix[Float], DenseMatrix[Float]] {
+case class LocalColumnPCAEstimator(dims: Int) extends Estimator[DenseMatrix[Float], DenseMatrix[Float]]
+  with SolverCostModel {
+
+  val pcaEstimator = new PCAEstimator(dims)
+
+  protected def fit(data: RDD[DenseMatrix[Float]]): Transformer[DenseMatrix[Float], DenseMatrix[Float]] = {
+    val singleTransformer = pcaEstimator.fit(data.flatMap(x => MatrixUtils.matrixToColArray(x)))
+    BatchPCATransformer(singleTransformer.pcaMat)
+  }
+
+  override def cost(
+    n: Long,
+    d: Int,
+    k: Int,
+    sparsity: Double,
+    numMachines: Int,
+    cpuWeight: Double,
+    memWeight: Double,
+    networkWeight: Double): Double = {
+    pcaEstimator.cost(n, d, k, sparsity, numMachines, cpuWeight, memWeight, networkWeight)
+  }
+}
+
+/**
+ * Estimates a PCA model for dimensionality reduction based on a sample of a larger input dataset,
+ * using a distributed PCA algorithm.
+ * Treats each column of the input matrices like a separate DenseVector input to [[DistributedPCAEstimator]].
+ *
+ * @param dims Dimensions to reduce input dataset to.
+ */
+case class DistributedColumnPCAEstimator(dims: Int) extends Estimator[DenseMatrix[Float], DenseMatrix[Float]]
+  with SolverCostModel {
```

Comment:
Just `CostModel` would suffice -- Also its an internal API so we can augment / change this in the future I guess